### ISSUE TITLE
TypeError: set_node_attributes() takes exactly 3 arguments (2 given)

### ISSUE THREAD

**Issue by BioSina** [networkx/networkx#3417]
I understand through the reference that set_node_attribute can take three arguments, but what I am trying to do is this:

w = G.node[c]['weight'] + 1
attrs = {c: {'weight': w}}
nx.set_node_attributes(G, attrs)

which should work fine, compared to the last example from the reference on the function:
>>>G = nx.path_graph(3)
>>> attrs = {0: {'attr1': 20, 'attr2': 'nothing'}, 1: {'attr2': 3}}
>>> nx.set_node_attributes(G, attrs)

**Comment by dschult** [comment]
For a single node, it is probably better to set the node attributes directly. This function is typically used to set many/all the nodes.  You could, for example do:

    G.nodes[c]['weight'] += 1

As far as the error message you are seeing, I suspect that the reference you are using is for a different version of NetworkX than the version you are using. Check the version using ```print(nx.__version__)```

**Comment by BioSina** [comment]
The reference is for version 2.2, which is the version I am using ( I just installed it on a new laptop and the first thing I tried to solve the issue on the server was updating). Stack overflow somewhere said that the access the way you describe it would be read-only and I have to use set_node_attributes() to change them, which is why I tried to do that. But it seems to work now. Still, the reference for 2.2 states what I tried should have worked as well.
